assign-profile.lua

```diff
@@ -92,36 +94,25 @@ local scripts = {
 
 local default_filename = "/hack/scripts/dwarf_profiles.json"
 
--- ----------------------------------------------- UTILITY FUNCTIONS ------------------------------------------------ --
-local function contains(table, value)
-    for _, v in pairs(table) do
-        if v == value then
-            return true
-        end
-    end
-    return false
-end
-
 -- ------------------------------------------------- APPLY PROFILE -------------------------------------------------- --
---- Apply the given profile to a unit, erasing or resetting the unit characteristics if requested.
----   :profile: nil, or a table. Each field has a characteristic name as key, and a table suitable to be passed as
----             an argument to the ``assign`` function of the module related to the characteristic.
----             See the modules documentation for more details.
----   :unit: a valid unit id, a df.unit object, or nil. If nil, the currently selected unit will be targeted.
----   :reset: nil, or a table value/boolean. See this script documentation for valid values.
+--- Apply the given profile to a unit, erasing or resetting the unit characteristics as requested.
+---   :profile: a table. Each field has a characteristic name as key, and a table suitable to be passed as
+---             an argument to the ``assign`` function of the script related to the characteristic.
+---             See the called script documentation for more details.
+---   :unit: a valid unit id, a df.unit object, or nil. The called script has the responsibility to validate this value.
+---   :reset: nil, or a list of values. See this script documentation for accepted values.
 --luacheck: in=string[],df.unit,bool[]
-function apply_profile(profile, unit, reset_table)
-    assert(not profile or type(profile) == "table")
+function apply_profile(profile, unit, reset)
+    assert(type(profile) == "table")
     assert(not unit or type(unit) == "number" or type(unit) == "userdata")
```

Comment:
The last check could be `df.unit:is_instance(unit)` - this is safe for non-userdata types as well.

Reply:
Fixed!
I should change it in my other scripts as well, so I will probably open a single PR with that single change for every script, otherwise I can open a PR for each script, as you prefer.

About the changelog: the main difference from the previous script is the way the list of "reset" options is passed while calling the script via the DFHack console (it now requires the arguments, if more than one, to be enclosed between square brackets). The documentation is changed accordingly, so if someone uses the old syntax to pass a list will receive an error, and in this case they will probably just go and check the documentation again to discover the new syntax. I don't think it's worth mentioning in the changelog.

Reply:
A single PR for that sort of thing (assuming the diff isn't massive) is fine with me.